### PR TITLE
Add yacht exercise

### PR DESCRIPTION
**Pull request by glennj**

### CODE REVIEW

**Review by kotp — APPROVED** [review]

**Comment by m-dango** [comment]
I want to have a think about this one for the categories arg, wondering if there's a better option than a string (thinking about how its done in [Raku](https://github.com/exercism/raku/blob/main/exercises/practice/yacht/t/yacht.rakutest) and [Python](https://github.com/exercism/python/blob/main/exercises/practice/yacht/yacht_test.py))

**Comment by glennj** [comment]
Perhaps something like
```perl
package Yacht::Categories;

# exports ...

use constant {
    ONES => 1,
    TWOS => 2,
    # ...
    CHOICE => 12,
};
```

Or, do away with the category argument, and the generator calls a subroutine named after the category:
```
properties:
   score:
     test: |-
       use Data::Dmp;
       sprintf(<<'END', $case->{input}{category}, dmp($case->{input}{dice}), $case->{expected}, dmp($case->{description}));
       is(
           %s(%s),
           %s,
           %s,
       );
       END
```

**Comment by glennj** [comment]
I added a Category submodule with Readonly values for the categories.

This is a bit awkward, as there's no way to stash it in the template-data.yaml file.

File: exercises/practice/yacht/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    ],
+    "test": [
+      "t/yacht.t"
+    ],
+    "example": [
+      ".meta/solutions/lib/Yacht.pm"
+    ],
+    "editor": [
+      "lib/Yacht/Category.pm"
+    ]
```
Comment (glennj) [inline exercises/practice/yacht/.meta/config.json:20]: Note this. This makes the Category module visible in the web editor, but readonly.

**Pull request merged**